src/presetsController.ts

```diff
@@ -1943,26 +1943,64 @@ export class PresetsController {
         return vscode.window.showTextDocument(vscode.Uri.file(presetsFilePath));
     }
 
+    // this is used for the file watcher on adding a new presets file
+    async onCreatePresetsFile() {
+        await this.reapplyPresets();
+        await this.project.projectController?.updateActiveProject(this.workspaceFolder);
+    }
+
     private async watchPresetsChange() {
-        if (this._presetsWatcher) {
-            this._presetsWatcher.close().then(() => {}, () => {});
-        }
 
-        const handler = () => {
+        const presetChangeHandler = () => {
             void this.reapplyPresets();
         };
-        this._presetsWatcher = chokidar.watch(this._referencedFiles, { ignoreInitial: true })
-            .on('add', handler)
-            .on('change', handler)
-            .on('unlink', handler);
+        const presetCreatedHandler = () => {
+            void this.onCreatePresetsFile();
+        };
+
+        const events: Map<string, () => void> = new Map<string, () => void>([
+            ["change", presetChangeHandler],
+            ["unlink", presetChangeHandler],
+            ["add", presetCreatedHandler]
+        ]);
+
+        this._presetsWatchers?.dispose();
+        this._presetsWatchers = new FileWatcher(this._referencedFiles, events, { ignoreInitial: true });
     };
 
     dispose() {
-        if (this._presetsWatcher) {
-            this._presetsWatcher.close().then(() => {}, () => {});
-        }
+        this._presetsWatchers?.dispose();
+
         if (this._sourceDirChangedSub) {
             this._sourceDirChangedSub.dispose();
         }
     }
 }
+
+/**
+ * FileWatcher is a wrapper around chokidar's FSWatcher that allows for watching multiple paths.
+ * Chokidar's support for watching multiple paths is currently broken, if it is fixed in the future, this class can be removed.
+ */
+class FileWatcher {
+    private watchers: Map<string, chokidar.FSWatcher>;
+
+    public constructor(paths: string | string[], eventHandlers: Map<string, () => void>, options?: chokidar.WatchOptions) {
+        this.watchers = new Map<string, chokidar.FSWatcher>();
+
+        for (const path of Array.isArray(paths) ? paths : [paths]) {
+            try {
+                const watcher = chokidar.watch(path, { ...options });
+                Array.from(eventHandlers).forEach(([event, handler]) => watcher.on(event, handler));
```

Comment:
One thing to consider here. When `.forEach` is called, the loop does not wait for all of the `forEach` methods to finish before moving to the next line. 

So for example, if it takes long enough to add all of the handlers, `this.watchers.set` may be called prior to all of the handlers being added. While I don't immediately see any issue arising from this, I worry whether the watcher would get set before the changes and then they don't get stored. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#description

The alternative would be to use a traditional for loop here. What do you think?